### ISSUE TITLE
How do the "Distributions" APIs work?

### ISSUE THREAD

**Issue by CycloneMAO** [tensorforce/tensorforce#802]
Hi Alex! I'm using 0.6.3, and cannot understand how do "Distributions" exactly work together with my customized networks. 

- The class `tensorforce.core.distributions.Gaussian` and `tensorforce.core.distributions.Categorical` have similar arguments such as `temperature_mode` and `stddev_mode`.  The value `predicted` will let the agent predict the standard deviation via a linear transformation of the state embedding. So does it mean that the outputs for both parameters are actually nodes forwarded from the **final layer** of the user-defined network (the other are the means for Gaussian or slopes for the categories)? But how does it guarantee positivity for both parameters? (via which activation function?) 
- Is `bounded_transform` for `tensorforce.core.distributions.Gaussian` just a `tanh` activation for the means (which will backpropagate its gradient to the whole network)? Or it doesn't connect to the network but serves as a transformation for the output action?
- Some agents have the parameter `exploration`. How does it work together with `tensorforce.core.distributions.Gaussian` or `tensorforce.core.distributions.Beta`? Does it actually generate random Gaussian variables with means equal to the action output(any `float` action) and standard deviation as specified by itself?
- BTW, I'm quite interested in the tensorboard functionality. However, I don't understand the example codes as following:
    ```
    Agent.create(...
        summarizer=dict(
            directory='data/summaries',
            # list of labels, or 'all'
            labels=['entropy', 'kl-divergence', 'loss', 'reward', 'update-norm']
        ), ...
    )
    ```
    Would you mind make it more specific? Thank you for your review.

**Comment by AlexKuhnle** [comment]
Hi @CycloneMAO,

- Yes, each of mean/stddev or logits/temperature (if `'predicted'`) are produced by an additional linear layer following the output of the user-specified network (hence, by the way, this network can output a vector of any size, since the final linear layer will turn it into the right size). The values for stddev and temperature are made positive by applying a slightly shifted softplus function (concretely: `y = (ln(1 + e^x) + 0.2) / (ln(2) + 0.2)`, so that `y=1` if `x=0` and the values don't get very small, currently roughly `>= 0.25`).
- `tanh` is applied to the means and transforms the distribution, so the gradients through the distribution back to the network will be affected accordingly. The other option, `'clipping'` does not do that, but simply clips the output actions.
- `exploration` is completely independent of the policy and any potential distributions involved. It simply adds noise to the output actions: Gaussian noise with stddev `exploration` in case of continuous actions, and randomly picking an action with probability `exploration` in case of categorical actions.
- This simply means that the agent records certain values `labels` as TensorBoard plots and stores the data in `directory`. You can visualize the plots via `tensorboard --logdir data/summaries` (in this case). Or am I misunderstanding the question? :-)

**Comment by AlexKuhnle** [comment]
I assume this can be closed.
